python2-pybind11/challenge/basics.cpp

```diff
@@ -0,0 +1,40 @@
+#include <pybind11/pybind11.h>
+
+#include "basics.hpp"
+
+namespace py = pybind11;
+
+PYBIND11_DECLARE_HOLDER_TYPE(T, std::shared_ptr<T>);
+
+namespace basics {
+
+PYBIND11_PLUGIN(basics) {
```

Comment:
I'm kind of surprised that putting the `PYBIND11_PLUGIN` macro call inside a namespace works, because it needs to generate a specific C symbol (exactly "initbasics") for module import to work.  I suppose `extern "C"` cancels out the namespace?

In any case, unless this is explicitly what the pybind11 docs suggest, I'd recommend moving it outside the namespace block.  It doesn't do any harm, but it's also a bit misleading, since the generated function isn't actually in the namespace (since it's `extern "C"`).